### ISSUE TITLE
Potential issues with generating shared libraries ABI in Windows 32bits

### ISSUE THREAD

**Issue by pedromsilvapt** [ziglang/zig#6092]
I'm running Windows 64bits, Zig 0.6, and wanted to build a simple DLL to be called from Python 32bits. However, calling a function and passing a struct by value somehow corrupts the data.

```zig
const std = @import("std");

const Value = extern struct {
    kind: u8,
    integer: i32 = 0,
    float: f64 = 0,
    size: usize = 0,
};

export fn test_create_struct(kind: u8, integer: i32) Value {
    return Value{ .kind = kind, .integer = integer };
}

export fn test_pass_struct(ptr: *u8, value: Value) void {
    std.debug.print("Test kind is {} {}\n", .{ value.kind, value.integer });
}
```

Which I then call from Python
```python
struct = lib.test_create_struct(1, 20)
lib.test_pass_struct(pointer(struct), struct)
```

The the output is incorrect (`Test kind is 20 0`, when it should really be `Test kind is 1 20`). But if I create the function and struct in C and compile with `gcc`, it works correctly.

If someone could try to reproduce the issue on their end, maybe with Python 64bits, or on Zig master, you can checkout this very simple and ready to use gist I created to reproduce the issue. It just requires Zig, GCC and Python: https://gist.github.com/pedromsilvapt/32ca3a412494d9aacc0c2f243470c6d2

I don't know if the problem is some issue in the Zig ABI, or if I'm doing something obviously wrong (which is entirely possible). Any help would be deeply appreciated.

**Comment by lithiumtoast** [comment]
Hello, I have a similar issue with calling functions from Zig shared library. The language I'm using however is C#. Calling the function in C# from a C built shared library does work as expected.

I created a demo repository here: https://github.com/lithiumtoast/pinvoke-playground

Zig: 0.6.0+3750cc06f (2020-08-28)
OS: macOS Catalina 10.15.6 (64-bit)

**Comment by Vexu** [comment]
x86 C ABI hasn't been implemented yet, #10852
